pkg/kubelet/cm/cpumanager/state/checkpoint.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"encoding/json"
+	"hash/fnv"
+
+	"k8s.io/kubernetes/pkg/kubelet/checkpointmanager"
+	"k8s.io/kubernetes/pkg/kubelet/checkpointmanager/errors"
+	hashutil "k8s.io/kubernetes/pkg/util/hash"
+)
+
+// cpuManagerCheckpoint struct is used to store cpu/pod assigments in a checkpoint
+type cpuManagerCheckpoint struct {
+	PolicyName string
+	DefaultCPUSet string
+	Entries map[string]string
+	Checksum uint64
+}
+
+// NewCPUManagerCheckpoint returns an instance of Checkpoint
+func NewCPUManagerCheckpoint() checkpointmanager.Checkpoint {
+	return &cpuManagerCheckpoint{Entries: make(map[string]string)}
+}
+
+// MarshalCheckpoint returns marshalled checkpoint
+func (cp *cpuManagerCheckpoint) MarshalCheckpoint() ([]byte, error) {
+	return json.Marshal(*cp)
+
+}
+// UnmarshalCheckpoint tries to unmarshal passed bytes to checkpoint
+func (cp *cpuManagerCheckpoint) UnmarshalCheckpoint(blob []byte) error {
+	if err := json.Unmarshal(blob, cp); err != nil {
+		return err
+	}
+	return cp.VerifyChecksum()
+}
+
+// GetChecksum returns calculated checksum of checkpoint
+func (cp *cpuManagerCheckpoint) GetChecksum() uint64 {
+	orig := cp.Checksum
+	cp.Checksum = 0
+	hash := fnv.New32a()
+	hashutil.DeepHashObject(hash, *cp)
+	cp.Checksum = orig
+	return uint64(hash.Sum32())
+}
+
+// VerifyChecksum verifies that current checksum of checkpoint is valid
+func (cp *cpuManagerCheckpoint) VerifyChecksum() error {
```

Comment:
Is this function used? It isn't part of `checkpointmanager.Checkpoint` interface, it seems to be declared in all implementations that I am aware of but we can't find any piece of code calling it.

Reply:
It is only used in `UnmarshalCheckpoint`, but I agree that it's not needed to put this functionality in a separate function as there probably won't be a case when it will be needed outside of `UnmarshalCheckpoint`